src/components/course-listing.cjsx

```diff
@@ -0,0 +1,76 @@
+_ = require 'underscore'
+React = require 'react'
+BS = require 'react-bootstrap'
+Router = require 'react-router'
+
+{CourseListingActions, CourseListingStore} = require '../flux/course-listing'
+
+# Called once the store is loaded
+# checks the course and roles and will redirect if there is only a single course and role
+DisplayOrRedirect = (transition, callback) ->
+  courses = CourseListingStore.allCourses() or []
+  if courses.length is 1 and courses[0].roles?.length is 1
```

Comment:
Hm, I don't think that'll work.  `courses` starts out as an array that may be empty.  Your `[courses]` statement converts it to a nested array but doesn't check if it's empty.